CMakeLists.txt

```
@@ -55,22 +57,27 @@ endif()
 find_package(Qt5 5.9 REQUIRED Network Gui Multimedia Test)
 get_filename_component(Qt5_Prefix "${Qt5_DIR}/../../../.." ABSOLUTE)
 
-if ((NOT DEFINED USE_INTREE_LIBQOLM OR USE_INTREE_LIBQOLM)
-        AND EXISTS ${PROJECT_SOURCE_DIR}/3rdparty/libQtOlm/lib/utils.h)
-    add_subdirectory(3rdparty/libQtOlm EXCLUDE_FROM_ALL)
-    include_directories(3rdparty/libQtOlm)
-    if (NOT DEFINED USE_INTREE_LIBQOLM)
-        set (USE_INTREE_LIBQOLM 1)
+if (${PROJECT_NAME}_ENABLE_E2EE)
+    add_definitions(-DE2EE_ENABLED)
```

I'd prefer to rather see [`target_compile_definitions`](https://cmake.org/cmake/help/v3.10/command/target_compile_definitions.html#command:target_compile_definitions) here, something like
```suggestion
    target_compile_definitions(${PROJECT_NAME} PUBLIC E2EE_ENABLED)
```
Don't have an opinion on whether the preprocessor symbol should be namespaced, like `QUOTIENT_E2EE_ENABLED`.

---

Not sure either, decided to stick with `QUOTIENT_E2EE_ENABLED`